doc/did-method-spec.md

```diff
@@ -185,39 +188,58 @@ DID document with type `EcdsaSecp256k1RecoveryMethod2020` and an `blockchainAcco
 Each identity always has a controller address. By default, it is the same as the identity address, but check the read
 only contract function `identityOwner(address identity)` on the deployed version of the ERC1056 contract.
 
-The identity controller will always have a `verificationMethod` entry with the id set as the DID with the fragment `#controller`
-appended.
+The identity controller will always have a `verificationMethod` entry with the id set as the DID with the fragment
+`#controller` appended.
 
 An entry for the controller is also added to the `authentication` array of the DID document.
 
 #### Enumerating Contract Events to build the DID Document
 
 The ERC1056 contract publishes three types of events for each identity.
 
-- `DIDOwnerChanged` (indicating a change of controller)
+- `DIDOwnerChanged` (indicating a change of `controller`)
 - `DIDDelegateChanged`
 - `DIDAttributeChanged`
 
 If a change has ever been made for an identity the block number is stored in the changed mapping.
 
 The latest event can be efficiently looked up by checking for one of the 3 above events at that exact block.
 
-Each event contains a `previousChange` value which contains the block number of the previous change (if any).
+Each ERC1056 event contains a `previousChange` value which contains the block number of the previous change (if any).
 
-To see all changes in history for an identity use the following pseudo code:
+To see all changes in history for an identity use the following pseudo-code:
 
-1. Call `changed(address identity)` on the ERC1056 contract.
+1. eth_call `changed(address identity)` on the ERC1056 contract to get the latest block where a change occurred.
 2. If result is `null` return.
 3. Filter for events for all the above types with the contracts address on the specified block.
 4. If event has a previous change then go to 3
 
-#### Delegate Keys
+After building the history of events for an address, interpret each event to build the DID document like so: 
+
+##### Controller changes (`DIDOwnerChanged`)
+
+When the controller address of a `did:ethr` is changed, a `DIDOwnerChanged` event is emitted.
+
+```solidity
+event DIDOwnerChanged(
+  address indexed identity,
+  address owner,
```

Comment:
maybe, we should say something about owner vs controller. that owner is basically the subject of the did doc and the controller if not other specified.

Reply:
good point.
I think it is more like
> `identity` address is the subject of the DID doc, and `owner` address represents the current `controler`, in DID terms.
> In case the `owner` address has never been changed, then the `identity` address is the implicit `controller`

Reply:
I added a section specifically about this